python/pyspark/sql/functions.py

```diff
@@ -3118,37 +3118,34 @@ def pandas_udf(f=None, returnType=None, functionType=None):
+    # |                        float|                  None|               1.0|               1.0|               1.0|                 1.0|                 1.0|               1.0|               1.0|               1.0|               1.0|           1.0|           1.0|           1.0|                                  X|                                                    X|                X|                   X|                            X|             X|                X|                 X|          X|                               X|  # noqa
+    # |                       double|                  None|               1.0|               1.0|               1.0|                 1.0|                 1.0|               1.0|               1.0|               1.0|               1.0|           1.0|           1.0|           1.0|                                  X|                                                    X|                X|                   X|                            X|             X|                X|                 X|          X|                               X|  # noqa
+    # |                         date|                  None|                 X|                 X|                 X|datetime.date(197...|                   X|                 X|                 X|                 X|                 X|             X|             X|             X|               datetime.date(197...|                                 datetime.date(197...|                X|datetime.date(197...|                            X|             X|                X|                 X|          X|                               X|  # noqa
+    # |                    timestamp|                  None|                 X|                 X|                 X|                   X|datetime.datetime...|                 X|                 X|                 X|                 X|             X|             X|             X|               datetime.datetime...|                                 datetime.datetime...|                X|datetime.datetime...|                            X|             X|                X|                 X|          X|                               X|  # noqa
+    # |                       string|                  None|                ''|                ''|                ''|              '\x01'|              '\x01'|                ''|                ''|            '\x01'|            '\x01'|            ''|            ''|            ''|                                  X|                                                    X|              'a'|                   X|                            X|            ''|                X|                ''|          X|                               X|  # noqa
```

Comment:
Hmm, is it normal? 1(int8), 1(int16) were converted to `u'\x01'`. But it is empty string now. ` 1(int32)` was ` u'\x01'` too, now it's still `'\x01'`.

Reply:
Yea, I was thinking this is a bit odd. This PR just captures the current status and we don't properly support the case for now anyway.. `u` is fine because it was python 2 and now python 3

Reply:
Yeah, that is a bit strange.. Let me look into it and maybe file a JIRA

Reply:
I made https://issues.apache.org/jira/browse/ARROW-5682 to track